cidc_api/models/models.py

```diff
@@ -488,9 +523,37 @@ def insert(self, session: Session, commit: bool = True, compute_etag: bool = Tru
             )
 
         logger.info(
-            f"admin-action: {grantor.email} gave {grantee.email} the permission {self.upload_type} on {self.trial_id}"
+            f"admin-action: {grantor.email} gave {grantee.email} the permission {self.upload_type or 'all assays'} on {self.trial_id or 'all trials'}"
+        )
+
+        # If this is a permission granting the user access to all trials for
+        # a given upload type or all upload types for a given trial, delete
+        # any related trial-upload type specific permissions to avoid
+        # redundancy in the database and in conditional IAM bindings.
+        perms_to_delete = (
```

Comment:
Nit: i see the point of cleaning up IAM bindings, not absolutely sure about the db cleanup.

Reply:
I think it gets harder to reason about correctness when we don't enforce a 1-1 correspondence between database permission records and IAM permission.

Reply:
right, then what do you think on not implicitly deleting those (granted by someone for some reason) explicit permissions at all? i understand there's a limit on the IAM bindings though. anyways it's a minor thing.